File: python/pytest/web_selenium/pages/ajax_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class AjaxPage:

    def __init__(self, driver):
        self.driver = driver
        self.button = driver.find_element(By.ID, "button")
        # self.title = driver.find_element(By.ID, "title")

    def click(self):
        self.button.click()

    def verify(self):
        wait = WebDriverWait(self.driver, 15)
        wait.until(EC.presence_of_element_located((By.ID, "title")))
        title = self.driver.find_element(By.ID, "title")
        assert title.text == "AJAX"

File: nodejs/selenium/pages/ajax.page.ts
import { By, WebDriver, WebElement, until } from "selenium-webdriver";
import assert from "assert";

export class AjaxPage {
  driver: WebDriver;
  button!: WebElement;

  constructor(driver: WebDriver) {
    this.driver = driver;
  }

  async init(): Promise<this> {
    this.button = await this.driver.findElement(By.id("button"));
    return this;
  }

  async click(): Promise<void> {
    await this.button.click();
  }

  async verify_title(): Promise<void> {
    // Wait up to 15 seconds until element with ID "title" is present
    await this.driver.wait(until.elementLocated(By.id("title")), 15000);

    const title = await this.driver.findElement(By.id("title"));
    const text = await title.getText();

    assert.strictEqual(text, "AJAX", `Expected title text to be "AJAX" but got "${text}"`);
  }
}
